sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -322,7 +343,10 @@ typedef struct
  * @param[in] client The #az_iot_hub_client to use for this call.
  * @param[in] received_topic An #az_span containing the received topic.
  * @param[out] out_request If the message is a C2D request, this will contain the
- * #az_iot_hub_client_c2d_request
+ * #az_iot_hub_client_c2d_request.
+ * @pre \p client must not be `NULL`.
+ * @pre \p received_topic must be a valid span of size greater than 0.
+ * @pre \p out_request must not be `NULL`.
```

Comment:
Missing a precondition doc for the client internal host name. Was this skipped intentionally?
Not only must the client not be `NULL`, it needs to be initialized first by calling `az_iot_hub_client_init` (from https://github.com/Azure/azure-sdk-for-c/pull/1314)

https://github.com/Azure/azure-sdk-for-c/blob/4b2f787bc155d1cd67f64ad1eebfdbc1438f6ce5/sdk/src/azure/iot/az_iot_hub_client_c2d.c#L25

Reply:
good call